Applications/Utils/FileConverter/TecPlotTools.cpp

```diff
@@ -0,0 +1,410 @@
+            continue;
+        }
+        else if (line.find("ZONE") != std::string::npos)
+        {
+            if (dataCountError(out, name, val_count, val_total)) return -3;
+            writeTecPlotSection(out, file_name, write_count, val_count, val_total);
+            out << line << "\n";
+            name = getName(line);
+            dims = getDimensions(line);
+            val_total = dims.first * dims.second;
+            val_count = 0;
+            continue;
+        }
+
+        out << line << "\n";
+        val_count++;
+    }
+    if (dataCountError(out, name, val_count, val_total))
+        return -3;
+    INFO("Writing time step #%i", write_count);
+    out.close();
+    INFO("Finished split.");
+    return 0;
+}
+
+/// Converts a TecPlot file into one or more OGS-meshes (one mesh per section/zone)
+int convertFile(std::ifstream& in, std::string file_name)
+{
+    std::string line(""), name("");
+    std::pair<std::size_t, std::size_t> dims(0,0);
+    std::vector<std::string> var_names;
+    std::vector< std::vector<double> > scalars;
+    std::size_t val_count(0), val_total(0);
+    std::size_t write_count(0);
+    while (getline(in, line))
+    {
+        if (line.find("GEOMETRY") != std::string::npos)
+            skipGeometrySection(in, line);
+
+        if (line.empty())
+            continue;
+        else if (line.find("TITLE") != std::string::npos)
+        {
+            if (dataCountError(name, val_count, val_total)) return -3;
+            if (val_count != 0)
+            {
+                writeDataToMesh(file_name, write_count, var_names, scalars, dims);
+                resetDataStructures(var_names.size(), scalars, val_count);
+            }
+            continue;
+        }
+        if (line.find("VARIABLES") != std::string::npos)
+        {
+            if (val_count != 0)
+            {
+                if (dataCountError(name, val_count, val_total)) return -3;
+                writeDataToMesh(file_name, write_count, var_names, scalars, dims);
+            }
+            var_names.clear();
+            var_names = getVariables(line);
+            resetDataStructures(var_names.size(), scalars, val_count);
+            continue;
+        }
+        if (line.find("ZONE") != std::string::npos)
+        {
+            if (val_count != 0)
+            {
+                if (dataCountError(name, val_count, val_total)) return -3;
+                writeDataToMesh(file_name, write_count, var_names, scalars, dims);
+                resetDataStructures(var_names.size(), scalars, val_count);
+            }
+            name = getName(line);
+            dims = getDimensions(line);
+            val_total = dims.first * dims.second;
+            val_count = 0;
+            continue;
+        }
+
+        double x;
+        std::stringstream iss(line);
+        std::size_t i(0);
+        std::size_t const n_scalars (scalars.size());
+        while (iss >> x)
+        {
+            if (i > n_scalars-1)
```

Comment:
Is it possible that `n_scalars` is zero at this point? I'm not sure, but `n_scalars - 1` is probably very large. Is this okay?

Reply:
Barring an invalid TecPlot-file, ```n_scalars``` must be larger than ```0```, otherwise the ```zone```-section would contain no data.